entity-service/src/integrationTest/java/org/hypertrace/entity/service/service/EntityQueryServiceTest.java

```diff
@@ -639,6 +641,74 @@ public void testBulkUpdateWithLabels() {
+    Entity.Builder apiEntityBuilder1 =
+        Entity.newBuilder()
+            .setTenantId(TENANT_ID)
+            .setEntityType(EntityType.API.name())
+            .setEntityName("api1")
+            .putIdentifyingAttributes(
+                EntityConstants.getValue(ServiceAttribute.SERVICE_ATTRIBUTE_ID),
+                createAttribute(SERVICE_ID))
+            .putIdentifyingAttributes(
+                EntityConstants.getValue(ApiAttribute.API_ATTRIBUTE_NAME), createAttribute("api1"))
+            .putIdentifyingAttributes(
+                EntityConstants.getValue(ApiAttribute.API_ATTRIBUTE_API_TYPE),
+                createAttribute(API_TYPE));
+    apiEntityBuilder1.putAttributes(
+        apiAttributesMap.get(API_LABELS), createStringArrayAttribute(List.of("Label1")));
+
+    Entity entity1 = entityDataServiceClient.upsert(apiEntityBuilder1.build());
+
+    Entity.Builder apiEntityBuilder2 =
+        Entity.newBuilder()
+            .setTenantId(TENANT_ID)
+            .setEntityType(EntityType.API.name())
+            .setEntityName("api2")
+            .putIdentifyingAttributes(
+                EntityConstants.getValue(ServiceAttribute.SERVICE_ATTRIBUTE_ID),
+                createAttribute(SERVICE_ID))
+            .putIdentifyingAttributes(
+                EntityConstants.getValue(ApiAttribute.API_ATTRIBUTE_NAME), createAttribute("api2"))
+            .putIdentifyingAttributes(
+                EntityConstants.getValue(ApiAttribute.API_ATTRIBUTE_API_TYPE),
+                createAttribute(API_TYPE));
+    apiEntityBuilder2.putAttributes(
+        apiAttributesMap.get(API_LABELS), createStringArrayAttribute(List.of("Label2")));
+
+    Entity entity2 = entityDataServiceClient.upsert(apiEntityBuilder2.build());
+
+    BulkEntityArrayAttributeUpdateRequest request =
+        BulkEntityArrayAttributeUpdateRequest.newBuilder()
+            .build()
+            .newBuilder()
+            .setEntityType(EntityType.API.name())
+            .addAllEntityIds(Set.of(entity1.getEntityId(), entity2.getEntityId()))
+            .setAttribute(ColumnIdentifier.newBuilder().setColumnName(API_LABELS))
+            .setOperation(BulkEntityArrayAttributeUpdateRequest.Operation.OPERATION_ADD)
+            .addAllValues(
+                List.of(
+                    LiteralConstant.newBuilder()
+                        .setValue(
+                            org.hypertrace.entity.query.service.v1.Value.newBuilder()
+                                .setString("Label3")
+                                .build())
+                        .build(),
+                    LiteralConstant.newBuilder()
+                        .setValue(
+                            org.hypertrace.entity.query.service.v1.Value.newBuilder()
+                                .setString("Label4")
+                                .build())
+                        .build()))
+            .build();
+
+    assertDoesNotThrow(
```

Comment:
We should verify the results. Did the labels actually merge?

Reply:
Added